keps/sig-node/5474-enable-writable-cgroups/README.md

```diff
@@ -0,0 +1,505 @@
+- `k8s.io/kubernetes/pkg/kubelet/kuberuntime`: Security context conversion tests including WritableCgroups mapping
+- `k8s.io/pod-security-admission/policy`: Pod Security Standards policy enforcement tests
+- `k8s.io/kubernetes/pkg/apis/core/v1`:  API defaulting and conversion tests
+
+##### Integration tests
+
+- `TestWritableCgroupsSecurityContextValidation`: API server validation integration test ensuring Linux-only enforcement
+- `TestPodSecurityStandardsWritableCgroups`: Pod Security Standards admission controller integration test
+- `TestKubeletSecurityContextConversion`: Kubelet CRI conversion integration test
+
+##### e2e tests
+
+- `test/e2e_node/writable_cgroup_test.go`: Node E2E tests covering:
+  - Basic functionality (writable vs read-only cgroups)
+  - Multi-container pods with mixed settings
+  - Integration with other SecurityContext fields
+  - cgroup v2 requirement validation
+  - Runtime compatibility checks
+
+### Graduation Criteria
+
+#### Alpha (v1.35)
+
+- Feature implemented behind a feature gate `WritableCgroups`
+- Basic API, validation, and kubelet implementation complete
+- CRI API and Containerd Changes
+- Unit and integration tests implemented
+- Pod Security Standards integration complete
+- Node E2E tests passing on v2 systems
+
+#### Beta (v1.36)
+
+- Feature gate enabled by default
+- E2E tests stable and passing consistently
+- At least one runtime (containerd) supporting the feature in a released version
+- User feedback incorporated from alpha testing
+
+#### GA (v1.37)
+
+- Feature stable and ready for production use
+- Conformance tests implemented where applicable
+- Documentation completed
+
+### Upgrade / Downgrade Strategy
+
+Enable/disable the feature gate
+
+**Upgrade**: 
+- New field is optional and defaults to `nil` (no change in behavior)
+- Existing workloads continue to function without modification
+
+**Update Flow**:
+- `WritableCgroups` field is **immutable** after pod creation.
+- Changes to `WritableCgroups` require pod recreation (delete + create)
+
+**Downgrade**:
+
+*Two scenarios depending on downgrade type:*
+
+**Feature Gate Disabled (same Kubernetes version):**
+- New pods with `WritableCgroups: true` will have the field silently dropped to `nil`
+- Existing pods with `WritableCgroups: true` continue to work
+- No errors occur - field is accepted but ignored
+
+**True Version Downgrade (to Kubernetes version without WritableCgroups field):**
+- Pods with `WritableCgroups` field will be **rejected** with strict decoding error: `unknown field "spec.containers[0].securityContext.writableCgroups"`
+- Remove the field from pod specs before downgrading
+
+### Version Skew Strategy
+
+**kubelet vs Container Runtime**:
+- On unsupported runtimes, the feature is silently ignored because the WritableCgroups field is not used.
+- Runtime version check not enforced to avoid breaking existing setups
```

Comment:
Can you expand on this? What would it mean to be enforced? Is this new behavior or leveraging existing kubelet behavior? Any links to relevant files are appreciated